llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -19157,6 +19165,10 @@ bool SLPVectorizerPass::vectorizeStoreChains(BoUpSLP &R) {
       return true;
     if (V1->getValueOperand()->getType() != V2->getValueOperand()->getType())
       return false;
+    if (V1->getValueOperand()->getType()->getScalarSizeInBits() !=
+        V2->getValueOperand()->getType()->getScalarSizeInBits()) {
+      return false;
+    }
```

Comment:
Ah, no, the check upper exactly check for this, so no need for this extra check.

Reply:
Ah, yes, it checks the Type, not the TypeID.